src/messaging/ExchangeMgr.cpp

```diff
@@ -161,17 +170,50 @@ ExchangeContext * ExchangeManager::AllocContext(uint16_t ExchangeId, uint64_t Pe
 void ExchangeManager::DispatchMessage(const PacketHeader & packetHeader, const PayloadHeader & payloadHeader,
                                       System::PacketBufferHandle msgBuf)
 {
+    CHIP_ERROR err                          = CHIP_NO_ERROR;
     UnsolicitedMessageHandler * umh         = nullptr;
     UnsolicitedMessageHandler * matchingUMH = nullptr;
-    CHIP_ERROR err                          = CHIP_NO_ERROR;
+    ExchangeContext * ec                    = nullptr;
+    uint16_t protocolId                     = 0;
+    uint8_t messageType                     = 0;
+    bool sendAckAndCloseExchange            = false;
+
+    protocolId  = payloadHeader.GetProtocolID();
```

Comment:
initialization could be done initially (without an extra = 0 pass).

Could protocolId and messageType be const?

Reply:
Acked